mw-webapp/src/logic/waysTable/MentoringWaysTable.tsx

```diff
@@ -1,32 +1,122 @@
 import {useEffect, useState} from "react";
+import {Button} from "src/component/button/Button";
+import {Modal} from "src/component/modal/Modal";
+import {UserPreviewDAL} from "src/dataAccessLogic/UserPreviewDAL";
 import {WayPreviewDAL} from "src/dataAccessLogic/WayPreviewDAL";
 import {PropsWithUuid} from "src/logic/waysTable/OwnWaysTable";
-import {waysColumns} from "src/logic/waysTable/waysColumns";
+import {columnHelper, waysColumns} from "src/logic/waysTable/waysColumns";
 import {WaysTable} from "src/logic/waysTable/WaysTable";
+import {UserPreview} from "src/model/businessModelPreview/UserPreview";
 import {WayPreview} from "src/model/businessModelPreview/WayPreview";
 
+/**
+ * Removes mentor from Way and returns new User Preview
+ */
+const removeMentorFromWay = async (userUuid: string, wayPreview: WayPreview) => {
+  const userPreview = await UserPreviewDAL.getUserPreview(userUuid);
+  const mentoringWays = userPreview.mentoringWays.filter((wayUuid) => wayUuid !== wayPreview.uuid);
+  const newUserPreview = new UserPreview({...userPreview, mentoringWays});
+
+  await UserPreviewDAL.updateUserPreview(newUserPreview);
+
+  const mentors = wayPreview.mentors.filter((mentor) => mentor.uuid !== userUuid);
+  const newWayPreview = new WayPreview({...wayPreview, mentors});
+
+  await WayPreviewDAL.updateWayPreview(newWayPreview);
```

Comment:
Could we parallelize this requests (updates)?

Reply:
Done